client/src/SalaryPieGraph.js

```diff
@@ -0,0 +1,90 @@
+import d3 from 'd3'
+
+export default class SalaryPieGraph {
+
+  init (node) {
+    this.width = node.clientWidth
+    this.height = 480
+    this.radius = Math.min(this.width, this.height) / 2
+
+    this.arc = d3.svg.arc()
+        .outerRadius(this.radius - 10)
+        .innerRadius(0)
+
+    this.pie = d3.layout.pie()
+        .sort(null)
+        .value((d) => d.salary)
+
+    this.chart = d3.select(node).append('svg')
+        .attr('width', this.width)
+        .attr('height', this.height)
+      .append('g')
+        .attr('transform', 'translate(' + this.width / 2 + ',' + this.height / 2 + ')')
+  }
+
+  create (players) {
+    let data = []
+    players.forEach((player, index) => {
+      data.push({
+        name: player.name,
+        pos: index,
+        salary: player.salary
+      })
+    })
+
+    let g = this.chart.selectAll('.arc')
+        .data(this.pie(data))
+      .enter().append('g')
+        .attr('class', 'arc')
+
+    g.append('path')
+        .attr('d', this.arc)
+        .attr('class', (d) => 'green-' + d.data.pos)
+        .each(function (d) { this._current = d }) // store the initial angles
```

Comment:
I wasn't able to port the arcTween function yet as it needed to reach back out to a global on window which I couldn't figure out how to match. The switch looks fine as is so I should probably remove this too.